### PR TITLE
Fix weakness potions thrown by peaceful factions not affecting zombie villagers

### LINKED ISSUE #1380: Peaceful Faction players unable to cure zombie_villager
### Debug info
*Run `/f debug mini` and paste or screenshot all four lines here.* 
[19:59:13 INFO]: Tuinity version: git-Tuinity-"071c8ed" (MC: 1.15.2)
[19:59:13 INFO]: Server ID: 45240292-c2a7-4fac-afac-0030000017ae
[19:59:13 INFO]: Plugin version: 1.6.9.5-U0.5.10-b224
[19:59:13 INFO]: Java version: 11.0.6

### Logs
*Give use your logs by going to `logs/latest.log`.*
Not relevant

### Details
*Please explain the problem.*
Due to peaceful faction members being unable to use splash potions that would harm, they are unable to cure zombie villagers. The process involved with curing a zombie villager starts with throwing a splash potion of weakness. Perhaps a config option to allow this specific potion, or any other way around this. Removing vanilla mechanics due to being peaceful hurts. (I restrict the switch to peaceful to a 1 time choice to avoid abuse, raiding, then going peaceful etc)

### PR DESCRIPTION
**Pull request by dariasc**
Closes #1380.

### CODE REVIEW

File: src/main/java/com/massivecraft/factions/listeners/FactionsEntityListener.java [hunk 1]
```diff
@@ -211,7 +204,19 @@ public void onPotionSplashEvent(PotionSplashEvent event) {
...
+            if (fPlayer.getFaction().isPeaceful()) {
+                for (PotionEffect effect : event.getPotion().getEffects()) {
+                    if (effect.getType().equals(PotionEffectType.WEAKNESS)) {
+                        // make sure this weakness potion only affects zombie villagers
+                        for (LivingEntity target : event.getAffectedEntities()) {
+                            if (target.getType() != EntityType.ZOMBIE_VILLAGER) {
+                                event.setIntensity(target, 0);
+                            }
+                        }
+                        return;
```
Comment (mbax) [inline src/main/java/com/massivecraft/factions/listeners/FactionsEntityListener.java:216]: What if a potion has multiple effects? I believe we should be nuking all other effects to?

**Comment by mbax** [comment]
At long last, pulled for next release <3

**Comment by dariasc** [comment]
:)

Sorry for not fixing your comment, my github notifications weren't arriving at my normal inbox...

**Comment by mbax** [comment]
> :)
> 
> Sorry for not fixing your comment, my github notifications weren't arriving at my normal inbox...

😀 

Made modifications so the result is now:

```java
            if (fPlayer.getFaction().isPeaceful()) {
                if (event.getPotion().getEffects().stream().allMatch(e -> e.getType() == PotionEffectType.WEAKNESS)) {
                    for (LivingEntity target : event.getAffectedEntities()) {
                        if (target.getType() != EntityType.ZOMBIE_VILLAGER) {
                            event.setIntensity(target, 0);
                        }
                    }
                    return;
                }
                event.setCancelled(true);
                return;
            }
```

This avoids the potential exploit of a weakness potion that also will murder the zombie.

**Pull request closed without merging**